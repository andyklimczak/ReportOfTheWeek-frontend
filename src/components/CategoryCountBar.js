import React, { PropTypes } from 'react';
import { BarChart, Bar, XAxis, Tooltip, Cell, YAxis } from 'recharts';
import { categoryToColor } from '../utils/utils.js';

const CategoryCountBar = ({ data }) => {
  return (
    <BarChart
      width={1500}
      height={400}
      data={data}>
      <Tooltip />
      <XAxis
        dataKey="category"
      />
      <YAxis
        dataKey="count"
      />
      <Bar
        barSize={60}
        name="Reviews"
        dataKey="count"
        minPointSize={5}>
        {
          data.map((datum, index) => <Cell fill={categoryToColor(datum.category)} key={index} />)
        }
      </Bar>
    </BarChart>
  );
};

CategoryCountBar.propTypes = {
  data: PropTypes.array
};

export default CategoryCountBar;
