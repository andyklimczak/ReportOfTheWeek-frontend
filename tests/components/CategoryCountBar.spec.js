import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import CategoryCountBar from '../../src/components/CategoryCountBar.js';

describe('CategoryCountBar', () => {
  const props = {
    data: [
      {
        product: 'Product 1',
        manufacturer: 'Manufacturer 1',
        category: 'Category 1',
        videoTitle: 'Video title 1',
        dateReleased: new Date(),
        rating: 7.5
      },
      {
        product: 'Product 2',
        manufacturer: 'Manufacturer 2',
        category: 'Category 2',
        videoTitle: 'Video title 2',
        dateReleased: new Date(),
        rating: 7.5
      },
      {
        product: 'Product 3',
        manufacturer: 'Manufacturer 3',
        category: 'Category 3',
        videoTitle: 'Video title 3',
        dateReleased: new Date(),
        rating: 7.5
      }
    ]
  };

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CategoryCountBar { ...props } />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(<CategoryCountBar { ...props } />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows three bars', () => {
    const element = mount(<CategoryCountBar { ...props } />);
    expect(element.find('.recharts-bar-rectangle').length).toBe(3);
  });
});
