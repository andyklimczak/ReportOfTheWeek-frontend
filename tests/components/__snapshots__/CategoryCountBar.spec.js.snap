exports[`CategoryCountBar snapshots 1`] = `
<div
  className="recharts-wrapper"
  onClick={[Function anonymous]}
  onMouseEnter={[Function anonymous]}
  onMouseLeave={[Function anonymous]}
  onMouseMove={[Function anonymous]}
  style={
    Object {
      "cursor": "default",
      "height": 400,
      "position": "relative",
      "width": 800
    }
  }>
  <svg
    className="recharts-surface"
    height={400}
    style={undefined}
    version="1.1"
    viewBox="0 0 800 400"
    width={800}>
    <g
      className="recharts-layer recharts-x-axis">
      <g
        className="recharts-layer recharts-cartesian-axis">
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={30}
          stroke="#666"
          width={790}
          x={5}
          x1={5}
          x2={795}
          y={365}
          y1={365}
          y2={365} />
        <g
          className="recharts-cartesian-axis-ticks">
          <g
            className="recharts-cartesian-axis-tick">
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={5}
              x1={136.66666666666666}
              x2={136.66666666666666}
              y={365}
              y1={371}
              y2={365} />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={136.66666666666666}
              y={371}>
              <tspan
                dy="0.71em"
                x={136.66666666666666}>
                Category 1
              </tspan>
            </text>
          </g>
          <g
            className="recharts-cartesian-axis-tick">
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={5}
              x1={400}
              x2={400}
              y={365}
              y1={371}
              y2={365} />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={400}
              y={371}>
              <tspan
                dy="0.71em"
                x={400}>
                Category 2
              </tspan>
            </text>
          </g>
          <g
            className="recharts-cartesian-axis-tick">
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={5}
              x1={663.3333333333333}
              x2={663.3333333333333}
              y={365}
              y1={371}
              y2={365} />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={663.3333333333333}
              y={371}>
              <tspan
                dy="0.71em"
                x={663.3333333333333}>
                Category 3
              </tspan>
            </text>
          </g>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-bar-graphical">
      <g
        className="recharts-layer recharts-bar">
        <g
          className="recharts-layer recharts-bar-rectangles">
          <g
            className="recharts-layer recharts-bar-rectangle" />
          <g
            className="recharts-layer recharts-bar-rectangle" />
          <g
            className="recharts-layer recharts-bar-rectangle" />
        </g>
      </g>
    </g>
  </svg>
</div>
`;
